src/laser_scan_assembler.cpp

```diff
@@ -117,16 +314,171 @@ class LaserScanAssembler : public BaseAssembler<sensor_msgs::LaserScan>
+    cv::Mat sorted_ranges = sortMatRowBy(height_values_, cropped_buffer, sorted_heights, reordering);
+
+    // Generate y_map for use in cv::remap
+    ROS_DEBUG("Create y_map_column");
+    cv::Mat y_map_column = cv::Mat::zeros(current_req_.vertical_resolution, 1, CV_32FC1);  // We'll stretch it later, as all the values in a row are the same for Y
+    auto vertical_step = fabs(current_req_.max_height - current_req_.min_height) / current_req_.vertical_resolution;
+    for (size_t y = 0; y < current_req_.vertical_resolution; y++)
+    {
+      auto height_at_pixel = current_req_.min_height + (y*vertical_step);
+      auto row_in_buffer = findInterpolatedIndex(sorted_heights, height_at_pixel);
+      // current_req_.vertical_resolution - y because we want the pixels with lower height at the bottom of the image of course
+      y_map_column.at<float>(current_req_.vertical_resolution - y, 0) = (float)row_in_buffer;
+      // ROS_INFO_STREAM("y: " << y <<
+      //   ", height_at_pixel: " << height_at_pixel <<
+      //   ", row_in_buffer:" << (float)row_in_buffer);
+    }
+    cv::Mat y_map;
+    cv::resize(y_map_column, y_map, cv::Size(current_req_.horizontal_resolution, current_req_.vertical_resolution), cv::INTER_NEAREST);
+
+    // Generate x map for use in cv::remap
+    ROS_DEBUG("Create x_map_row");
+    cv::Mat x_map_row = cv::Mat::zeros(1, current_req_.horizontal_resolution, CV_32FC1);  // We'll stretch it later, as all the values in a column are the same for X
+    auto horizontal_step = fabs(current_req_.max_width - current_req_.min_width) / current_req_.horizontal_resolution;
+    for (size_t x = 0; x < current_req_.horizontal_resolution; x++)
+    {
+      auto angle_at_pixel = current_req_.min_width + (x*horizontal_step);
+      auto column_in_buffer = (angle_at_pixel - last_scan_.angle_min) / last_scan_.angle_increment;
+      x_map_row.at<float>(0, x) = (float)column_in_buffer;
+    }
+    cv::Mat x_map;
+    cv::resize(x_map_row, x_map, cv::Size(current_req_.horizontal_resolution, current_req_.vertical_resolution), cv::INTER_NEAREST);
+
+    cv::Mat range_image = cv::Mat::zeros(current_req_.vertical_resolution, current_req_.horizontal_resolution, CV_16UC1);
+    ROS_DEBUG("Apply remapping");
+    cv::remap(sorted_ranges, range_image, x_map, y_map, cv::INTER_LINEAR, cv::BORDER_CONSTANT, 0);
+
+    // Convert the range image to depth image by multiplying it by cos(angle)
+    // First we create an image containing the angle at each image column
+
+    // Angles, step 1: for each pixel, calculate the angle and take the cos(angle)
+    cv::Mat cos_angle_row = cv::Mat::zeros(1, current_req_.horizontal_resolution, CV_32FC1);
+    // Calculate cosine of all angles, in place
+    cos_angle_row.forEach<float>([this, horizontal_step](float &value, const int* position) -> void {
+        auto angle_at_pixel = current_req_.min_width + (position[1]*horizontal_step);
```

Comment:
Angles are a linspace of course, can use Eigen for that too:
```c++
Eigen::ArrayXd x = Eigen::ArrayXd::LinSpaced(dynamic_config.max_vhist, 0, dynamic_config.max_vhist - 1);
```
or something like that.